ocs_ci/ocs/utils.py

```diff
@@ -942,56 +957,67 @@ def run_must_gather(log_dir_path, image, command=None, cluster_config=None):
         image (str): must-gather image registry path
         command (str): optional command to execute within the must-gather image
         cluster_config (MultiClusterConfig): Holds specifc cluster config object in case of multicluster
+        silent (bool): True if silent mode
+        output_file (bool): True if direct whole output to file instead of printing it out to log (apply
+          only if silent is True).
+        skip_after_max_fail (bool): When max number failed attempts to collect MG reached, will skip
+            MG collection.
+        timeout (int): Max timeout to wait for MG to complete before aborting the MG execution.
 
     Returns:
         mg_output (str): must-gather cli output
 
     """
-    # Must-gather has many changes on 4.6 which add more time to the collection.
-    # https://github.com/red-hat-storage/ocs-ci/issues/3240
+    global mg_fail_count, mg_last_fail, mg_collected_logs
+
+    max_mg_fail_attempts = config.REPORTING.get("max_mg_fail_attempts")
+    if skip_after_max_fail:
+        with mg_lock:
+            if mg_fail_count > max_mg_fail_attempts:
```

Comment:
are we deleting the MG dir structure in case of timeout failure?

Reply:
No, it's still contain some useful data produced by MG even if it time out.

Reply:
should we maybe delete the dir structure of failed MG collection, in case MG collection succeeds in a later attempt?

Reply:
there is a directory for every failed test case - we are not doing any retry for one failed connection we just continue to next test case - and next time another failure occur we are trying collect mg for another test failure. So new directory is created for MG and old one still have some valuable information - also only one of MG can fail - OCP or ODF - and we also collecting Noobaa logs to same directory - so really nothing to delete as all collected data are valuable if we have some.